Documentation/internals/hooks.rst

```diff
@@ -0,0 +1,16 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+eBPF Program Types
+==================
+
+Cilium uses the following eBPF program types to attach programs to the kernel:
+
+- ``BPF_PROG_TYPE_XDP``
+- ``BPF_PROG_TYPE_SCHED_ACT``
+- ``BPF_PROG_TYPE_CGROUP_SOCK_ADDR``
+- ``BPF_PROG_TYPE_SOCK_OPS``
```

Comment:
I'm not sure if we use the `*SOCK_OPS` or `SK_MSG` hooks.

Reply:
We still have sockmap code in the tree, so even if it's not always enabled, it is possible for Cilium to use these hooks.

Reply:
About that, should there be a distinction between hooks that we're using all the time (Is there any?) and those that are feature-dependant?